components/uaa-activator/internal/scheduler/scheduler.go

```diff
@@ -0,0 +1,60 @@
+package scheduler
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+// Scheduler takes steps and executes them
+type Scheduler struct {
+	logger *zap.Logger
+}
+
+// New returns a new scheduler
+func New(logger *zap.Logger) *Scheduler {
+	return &Scheduler{
+		logger: logger,
+	}
+}
+
+// Defines contract for scheduler
+type (
+	StepFunc func(ctx context.Context) error
+
+	Step struct {
+		Name string
+		Do   StepFunc
+	}
+
+	Steps []Step
+)
+
+// MustExecute takes steps and executes them
+func (s *Scheduler) MustExecute(ctx context.Context, steps Steps) {
```

Comment:
IMO, method name sounds like it was going to return boolean type (something must be executed or not). Why just not "ExecuteSteps"?

Reply:
discussed f2f. `MustExecute` can stay as it is :)